test/modules/http/compliance/http-res-destroy.test.ts

```diff
@@ -12,57 +14,158 @@ const httpServer = new HttpServer((app) => {
 const interceptor = new ClientRequestInterceptor()
 
 beforeAll(async () => {
-  interceptor.apply()
   await httpServer.listen()
 })
 
 afterEach(() => {
   interceptor.removeAllListeners()
+  interceptor.dispose()
 })
 
 afterAll(async () => {
-  interceptor.dispose()
   await httpServer.close()
 })
 
-it('emits the "error" event when a bypassed response is destroyed', async () => {
-  const socketErrorListener = vi.fn()
+const scenarios = [
+  {
+    condition: 'when the interceptor is not enabled [baseline test]',
+    setup: () => {},
+  },
```

Comment:
The first scenario is for plain Node.js without this library enabled at all. This means that we can check that the assertions we're using are correct based on the behaviour of Node.js.